text/0015-gas-payments.md

```diff
@@ -0,0 +1,90 @@
+# Casper Payments: Gas Payment Concepts
+
+## High level view
+This is the first of a few CEPs on Casper Payment model. 
+The motivation for this work is to make Casper much more user and dev 
+friendly than it is now. There are a few topics here, that mix. The change 
+to one of them might cause the domino effect on others, but we'll 
+try to chunk them into logical units.
+
+#### Token Standard 
+Most of the systems built on top of smart contracting platforms use tokens 
+and it is reasonable to put a lot of effort in designing a good token standard 
+or discover gaps in core functionalities, that should be fixed before 
+the Mainnet. The most popular token standard right now is the ERC20. We have 
+implemented its Casper version here https://github.com/casperlabs/erc20.
+We think we can do better and offer our future community a token standard 
+that avoids the "approve and call" pitfall. 
+This work will be the subject of future CEP.
+
+#### CSPR Interface
+One of the most important lessons we (as a blockchain community) learned 
+from the Ethereum is that it's problematic when the platform's native token 
+doesn't implement a good token standard. After our Token Standard is defined 
+we will consider CSPR reimplementation to support the Standard.
+This work will be the subject of future CEP.
+
+#### Gas Payments
+This part covers the topic of how users pay for the gas. It doesn't cover 
+gas model itself. This work is a subject for current CEP.
+
+#### Smart contract and accounts unification
+The new model of our accounting/operational unit that should easily allow for 
+all above. That means merging accounts and contracts (and urefs?) addressing 
+spaces, enabling easy transfer of tokens between accounts and addresses, 
+making uref sharing easy and ways of calling contracts.
+This work will be the subject of future CEP.
+
+## Gas Payments
+In this section, we'll cover ways of paying for deploys that were discussed
+in a past and we'll decide what we want to support.
+
+### Self Payment
+This is the simplest case. User should be able to use its own CSPRs to pay
+for the execution of its own deploy (WASM or contract call).
+
+### Delegeted Payment
```

Comment:
This is a good start, but maybe there should be a whole CEP dedicated specifically to Delegated Payments? It's going to be a complex issue to describe in the full CEP format, which present document does not follow.

Reply:
The high level proposition of delegated payment + removal of payment codes should be enough to catch obvious problems and estimate the work required. The detail spec should be done in cooperation with a core team member, that knows the code internals. If the CEP has enough support, I'll spend more time on it.